src/extensions/ThumbnailsExtension/ThumbnailsExtension.ts

```diff
@@ -43,6 +42,43 @@ export class ThumbnailsExtension extends Module {
       .sort((a, b) => b.start - a.start);
   }
 
+  private async loadBifThumbs(file) {
+    const response = await fetch(file);
+    const data = await response.arrayBuffer();
+
+    const BifParser = new BIFParserClass(data);
+    const imgs = BifParser.getImageData();
+
+    this.thumbnails = imgs
+      .map(item => {
+        const { start, src, x, y, width, height } = item;
+        return {
+          start,
+          src,
+          x,
+          y,
+          width,
+          height,
+        };
+      })
+      .sort((a, b) => b.start - a.start);
+  }
+
+  public async load() {
+    const file = this.instance.config.thumbnails.src;
+    // Get the file extension for conditional processing
+    const ext = file.split(".").pop();
+
+    if (ext === "vtt") {
+      this.loadVttThumbs(file);
+    } else if (ext === "bif") {
+      this.loadBifThumbs(file);
+    } else {
+      // We shouldn't get here, but still
+      console.warn("Invalid file type passed for thumbnails. Acceptable file types: vtt, bif");
```

Comment:
indigo has a logger, see `this.instance.log('ThumbnailsExtension')('Error message')`